### ISSUE TITLE
Cursor can disappear during movements in certain scenarios

### ISSUE THREAD

**Issue by dstein64** [neovim/neovim#25502]
### Problem

I noticed that as of PR #25395, the cursor can disappear during movements in certain scenarios. Here's a minimal reproducible example.

I'm not sure if the behavior is expected/intended in certain scenarios, so I thought I'd report it here.

Cc: @bfredl 

### Steps to reproduce

```
nvim --clean
```

Create `example.lua`, with the following text. Be sure to insert many (e.g., 100) blank lines at the end of the file, to have more room for cursor movement in the steps that follow.

```lua
vim.api.nvim_create_autocmd('CursorMoved', {
  callback = function()
    vim.defer_fn(function()
      for i = 0, 20 do
	vim.o.textwidth = 80
      end
    end, 0)
  end
})

vim.api.nvim_create_autocmd('OptionSet', {
  callback = function()
    vim.cmd('redrawstatus!')
  end
})

-- <INSERT MANY BLANK LINES>
```

Source the file:

```vim
:source %
```

Vertically split the window:

```vim
:vsplit
```

Go to the top of the file:

```vim
gg
```

Then hold `j` to move the cursor, which will disappear:

```vim
jjjjjjjjjjjjjjjjjjjjjjjjjjjjjjjjjjjj
```

### Expected behavior

I expected the cursor to show during movement, as was the case prior to PR #25395.

The following video shows cursor movement before and after sourcing the script. Before, the cursor shows during movement. After, the cursor disappears.

https://github.com/neovim/neovim/assets/541289/e30e8aac-3f3d-42dc-9000-360e8d48e7f6

### Neovim version (nvim -v)

NVIM v0.10.0-dev-1242+ge33269578

### Vim (not Nvim) behaves the same?

N/A

### Operating system/version

Kubuntu 23.04

### Terminal name/version

Konsole 22.12.3

### $TERM environment variable

xterm-256color

### Installation

build from repo

**Comment by dstein64** [comment]
The `example.lua` script above can be simplified. The `redrawstatus` call can be moved to the `CursorMoved` autocmd, and the `OptionSet` autocmd can be removed.

```lua
vim.api.nvim_create_autocmd('CursorMoved', {
  callback = function()
    vim.defer_fn(function()
      for i = 0, 20 do
        vim.o.textwidth = 80
        vim.cmd('redrawstatus!')
      end
    end, 0)
  end
})

-- <INSERT MANY BLANK LINES>
```

**Comment by dstein64** [comment]
As can be seen in the video posted above, it looks like the cursor moves to the right split status line, on top of the "t" in "tmp6.lua" (as opposed to fully _disappearing_, as I had reported).

**Comment by zeertzjq** [comment]
`:redrawstatus!` can be changed to `:redraw`.

**Comment by zeertzjq** [comment]
This seems to be a performance regression. However, whether the cursor "move"s to the statusline of the bottom-right window seems to depends on build flags.

**Comment by zeertzjq** [comment]
Fixed by #2553
